dali/test/python/test_utils.py

```diff
@@ -177,22 +195,47 @@ def compare_pipelines(
         compare_layouts (bool, optional): Whether to compare layouts of outputs between pipelines.
             Defaults to True.
     """
-    pipe1.build()
-    pipe2.build()
-    for _ in range(N_iterations):
+    for _ in range(n_iterations):
         out1 = pipe1.run()
         out2 = pipe2.run()
         assert len(out1) == len(out2)
         for i in range(len(out1)):
-            out1_data = out1[i].as_cpu() if isinstance(out1[i][0], dali.backend_impl.TensorGPU) else out1[i]
-            out2_data = out2[i].as_cpu() if isinstance(out2[i][0], dali.backend_impl.TensorGPU) else out2[i]
+            out1_data = out1[i].as_cpu() if isinstance(out1[i][0], dali.backend_impl.TensorGPU) \
+                else out1[i]
+            out2_data = out2[i].as_cpu() if isinstance(out2[i][0], dali.backend_impl.TensorGPU) \
+                else out2[i]
             if isinstance(expected_layout, tuple):
                 current_expected_layout = expected_layout[i]
             else:
                 current_expected_layout = expected_layout
             check_batch(out1_data, out2_data, batch_size, eps, max_allowed_error,
                         expected_layout=current_expected_layout, compare_layouts=compare_layouts)
-    print("OK: ({} iterations)".format(N_iterations))
+    print("OK: ({} iterations)".format(n_iterations))
+
+
+def compare_pipelines(pipe1, pipe2, batch_size, N_iterations, eps=1e-07, max_allowed_error=None,
```

Comment:
Can't you just add `pipe1.build()` to the original `compare_pipelines` and get rid of `compare_built_pipelines`?
If `build` has been called already pipeline will just treat it as a noop.